components/brave_origin/brave_origin_service.cc

```diff
@@ -0,0 +1,444 @@
+  }
+
+  // Check if this is a valid BraveOrigin preference
+  const BraveOriginPrefInfo* pref_info = GetPrefInfo(pref_name);
+  if (!pref_info) {
+    return false;
+  }
+
+  if (!local_state_) {
+    return false;
+  }
+
+  // Check if the pref is externally managed
+  const base::Value::Dict& policies_dict =
+      local_state_->GetDict(prefs::kBravePolicies);
+
+  std::string externally_managed_key;
+  if (pref_info->scope == BraveOriginPolicyScope::kGlobal) {
+    externally_managed_key = pref_name + prefs::kExternallyManagedSuffix;
+  } else {
+    externally_managed_key =
+        GetLocalStatePolicyPrefKey(profile_id_, pref_name) +
+        prefs::kExternallyManagedSuffix;
+  }
+
+  const base::Value* managed_value = policies_dict.Find(externally_managed_key);
+  bool is_externally_managed =
+      managed_value && managed_value->is_bool() && managed_value->GetBool();
+
+  // BraveOrigin controls it if it's NOT externally managed
+  return !is_externally_managed;
+}
+
+bool BraveOriginService::WasManagedBeforeBraveOrigin() const {
+  if (!local_state_) {
+    return false;
+  }
+
+  // Check if any policies were externally managed by looking for
+  // .externally_managed flags in local state
+  const base::Value::Dict& policies_dict =
+      local_state_->GetDict(prefs::kBravePolicies);
+
+  // Check both global and profile-scoped preferences
+  const auto& prefs = GetPolicyDefinitionsInstance();
+  for (const auto& [pref_name, pref_info] : prefs) {
+    std::string externally_managed_key;
+    if (pref_info.scope == BraveOriginPolicyScope::kGlobal) {
+      externally_managed_key = pref_name + prefs::kExternallyManagedSuffix;
+    } else {
+      externally_managed_key =
+          GetLocalStatePolicyPrefKey(profile_id_, pref_name) +
+          prefs::kExternallyManagedSuffix;
+    }
+
+    const base::Value* managed_value =
+        policies_dict.Find(externally_managed_key);
+    if (managed_value && managed_value->is_bool() && managed_value->GetBool()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void BraveOriginService::SetPolicyRefreshDelegate(
+    std::unique_ptr<PolicyRefreshDelegate> delegate) {
+  policy_refresh_delegate_ = std::move(delegate);
+}
+
+bool BraveOriginService::SetBraveOriginPolicyValue(const std::string& pref_name,
+                                                   base::Value value) {
+  const BraveOriginPrefInfo* pref_info = GetPrefInfo(pref_name);
+  if (!pref_info) {
+    return false;
+  }
+
+  // Set preferences based on their scope (user prefs)
+  if (value.is_bool()) {
+    // Only set user pref if it's user settable
+    if (pref_info->scope == BraveOriginPolicyScope::kGlobal &&
+        pref_info->user_settable && local_state_) {
+      local_state_->SetBoolean(pref_name, value.GetBool());
+    } else if (pref_info->scope == BraveOriginPolicyScope::kProfile &&
+               pref_info->user_settable && profile_prefs_) {
+      profile_prefs_->SetBoolean(pref_name, value.GetBool());
```

Comment:
I might be missing something, but why do we set the user pref here? Shouldn't we only be calling `UpdateLocalStatePolicyPref`?

Reply:
We keep the user prefs in sync on purpose